stats/opentelemetry/metricsregistry_test.go

```diff
@@ -0,0 +1,311 @@
+			},
+		},
+		{
+			Name:        "float-histo",
+			Description: "histogram of call values from tests",
+			Unit:        "float",
+			Data: metricdata.Histogram[float64]{
+				DataPoints: []metricdata.HistogramDataPoint[float64]{
+					{
+						Attributes:   attribute.NewSet(attribute.String("float histo label key", "float histo label value"), attribute.String("float histo optional label key", "float histo optional label value")),
+						Count:        1,
+						Bounds:       bounds,
+						BucketCounts: []uint64{0, 1, 0, 0},
+						Min:          metricdata.NewExtrema(float64(4.3)),
+						Max:          metricdata.NewExtrema(float64(4.3)),
+						Sum:          4.3,
+					},
+				},
+				Temporality: metricdata.CumulativeTemporality,
+			},
+		},
+		{
+			Name:        "simple-gauge",
+			Description: "the most recent int emitted by test",
+			Unit:        "int",
+			Data: metricdata.Gauge[int64]{
+				DataPoints: []metricdata.DataPoint[int64]{
+					{
+						Attributes: attribute.NewSet(attribute.String("int gauge label key", "int gauge label value")), // No optional label, not float.
+						Value:      8,
+					},
+				},
+			},
+		},
+	}
+
+	clientReader := otelmetric.NewManualReader()
+	clientProvider := otelmetric.NewMeterProvider(otelmetric.WithReader(clientReader))
+
+	// This configures the defaults alongside int counter 3. All the instruments
+	// registered except int counter 2 and 3 are default, so all measurements
+	// recorded should show up in reader collected metrics except those for int
+	// counter 2.
+	// This also only toggles the float count and float histo optional labels,
+	// so only those should show up in metrics emissions. All the required
+	// labels should show up in metrics emissions.
+	mo := MetricsOptions{
+		Metrics:        DefaultMetrics().Add("int-counter-3"),
+		OptionalLabels: []string{"float counter optional label key", "float histo optional label key"},
+		MeterProvider:  clientProvider,
+	}
+	// Don't have access to clientStatsHandler in DialOption, so just create
+	// one directly.
+	csh := &clientStatsHandler{options: Options{MetricsOptions: mo}}
+	csh.initializeMetrics()
+
+	serverReader := otelmetric.NewManualReader()
+	serverProvider := otelmetric.NewMeterProvider(otelmetric.WithReader(serverReader))
+	mo = MetricsOptions{
+		Metrics:        DefaultMetrics().Add("int-counter-3"),
+		OptionalLabels: []string{"float counter optional label key", "float histo optional label key"},
+		MeterProvider:  serverProvider,
+	}
+	// Don't have access to serverStatsHandler in ServerOption, so just create
+	// one directly.
+	ssh := &serverStatsHandler{options: Options{MetricsOptions: mo}}
+	ssh.initializeMetrics()
+
+	tests := []struct {
+		metricsRecorder estats.MetricsRecorder
+		reader          *otelmetric.ManualReader
+	}{
+		{
+			metricsRecorder: csh,
+			reader:          clientReader,
+		},
+		{
+			metricsRecorder: ssh,
+			reader:          serverReader,
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
Prefer immediate `t.Run(___, func(t *testing.T) { /* body of loop */})` inside a loop like this.

Reply:
Ah right I forgot about this. Switched.